docs/design/adminconsole/INSTANCE-MANAGEMENT.md

```diff
@@ -0,0 +1,62 @@
+# Instance Management
+
+## Objective
+
+The goal of this spike is to enhance the endpoints for `Instance` in the Admin Console so that they can handle information flowing into `odsInstance`, `odsInstanceDerivatives`, and `odsInstanceContexts`. This includes processing the provided payload, persisting data to the relevant tables, and ensuring proper references are maintained.
+
+## Expected Payload
+
+```json
+{
+  "instanceId": 1,
+  "tenantId": 1,
+  "document": {
+    "name": "Instance #1 - 2024",
+    "instanceType": null,
+    "odsInstanceContexts": [
+      {
+        "id": 1,
+        "InstanceId": 1,
+        "contextKey": "schoolYearFromRoute",
+        "contextValue": "2024"
+      }
+    ],
+    "odsInstanceDerivatives": [
+      {
+        "id": 1,
+        "InstanceId": 2,
+        "derivativeType": "Read"
+      }
+    ]
+  }
+}
+```
+
+## Processing Details
+
+1. **OdsInstance**: The `OdsInstance` object will be extracted from the payload and persisted into the table `dbo.OdsInstance`.
+2. **OdsInstanceDerivatives and OdsInstanceContexts**: These entities will follow the same process:
+    - The objects will be extracted from the payload.
+    - Each will reference the newly created `OdsInstance`.
+    - The data will be persisted into their respective tables.
+
+Both `OdsInstanceDerivatives` and `OdsInstanceContexts` should maintain proper relationships by referencing the newly created `OdsInstance`.
```

Comment:
@dfernandez-gap @jagudelo-gap based on the conversation we just had, do you see that this information needs to be stored in a brand new table instead of storing directly in the existing tables?

Later in the process _the instance management worker_ will need to send a `POST /v2/odsInstances` request, which will save the new entry in the `OdsInstance` table. And, you also need to make sere that the Admin API can accept the derivative and context information.